sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -3,92 +3,78 @@ package certificates
 import (
 	"context"
 	"fmt"
+	"sync/atomic"
 
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/pkg/centralsensor"
 	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/logging"
-	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/pkg/uuid"
 	"github.com/stackrox/rox/sensor/common/centralcaps"
 	"github.com/stackrox/rox/sensor/common/message"
 )
 
 var (
-	log                            = logging.LoggerForModule()
-	ErrCertificateRequesterStopped = errors.New("certificate requester is stopped")
+	log = logging.LoggerForModule()
 )
 
 // Requester defines an interface for requesting TLS certificates from Central
 type Requester interface {
-	Start()
-	Stop()
 	RequestCertificates(ctx context.Context) (*Response, error)
+	DispatchResponse(response *Response)
 }
 
-// NewLocalScannerCertificateRequester creates a new local scanner certificate requester that communicates through
-// the specified channels and initializes a new request ID for reach request.
-// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
-// This assumes that the returned certificate requester is the only consumer of `receiveC`.
-func NewLocalScannerCertificateRequester(sendC chan<- *message.ExpiringMessage,
-	receiveC <-chan *central.IssueLocalScannerCertsResponse) Requester {
+type MsgToCentralFn func(ctx context.Context, msg *message.ExpiringMessage) error
+
+// NewLocalScannerCertificateRequester creates a new certificate requester for Local Scanner certificates
+// (Scanner V2 and Scanner V4).
+func NewLocalScannerCertificateRequester(msgToCentralFn MsgToCentralFn) Requester {
 	return newRequester[
 		*central.IssueLocalScannerCertsRequest,
 		*central.IssueLocalScannerCertsResponse,
 	](
-		sendC,
-		receiveC,
 		&localScannerMessageFactory{},
-		&localScannerResponseFactory{},
 		nil,
+		msgToCentralFn,
```

Comment:
To resolve "the blocker", lets not pass `msgToCentralFn` but create a local channel in the `newRequester`.

More comments starting with (solution to the blocker) will suggest how I would rewrite the problematic part.